### PR TITLE
Major overhaul of Urchin

### PR DESCRIPTION
**Pull request by AlDu2407**
- Added TailwindCSS to Urchin.
- Updated Makefile to download TailwindCSS (currently only Linux is supported)
- Refactored templates to share similar logic between templates.
- Redesigned UI for Urchin.
- Added layout page shared between all sub pages.
- Added dark mode support.
- Added new `renderHtml` method to reduce duplicate code in go.
- Added self-designed minimalistic 'sea urchin' favicon. :D
- Added missing sites `about` and `services`.
- Added `Not Found` handling if a resource is requested that is not present, displaying an error message while keeping the layout page.
- Restructured `static` folder to include stylesheets, scripts and assets.

### CODE REVIEW

**Comment by matheusgomes28** [comment]
Hey @AlDu2407 , I wonder if you ever got a chance to add `wget` to the image?

Btw I think it has `curl`, so you could work around downloading the stuff you need on CI:

```
cp -r migrations tests/helpers/
if [ ! -f tailwindcss ]; then \
	wget -q https://github.com/tailwindlabs/tailwindcss/releases/latest/download/tailwindcss-linux-x64 \
		&& chmod +x tailwindcss-linux-x64 \
		&& mv tailwindcss-linux-x64 tailwindcss; \
fi
/bin/sh: 2: wget: not found
make: *** [Makefile:3[7](https://github.com/matheusgomes28/urchin/actions/runs/11663344344/job/32471628007?pr=97#step:4:8): install-tailwindcss] Error 127
```

**Comment by AlDu2407** [comment]
Hi @matheusgomes28 , 

is there a way for you to remove the line in the Dockerfile, that removes wget from the build image? I think afterwards the pipeline will succeed. And afterwards you should be able to review the entire code.

File: Makefile [hunk 1]
```diff
@@ -26,9 +27,17 @@ clean:
...
+# TODO: For now we support only the linux version of tailwindcss, has to be updated in the future to support Windows and MacOS as well.
 install-tools:
 	go install github.com/pressly/goose/v3/cmd/goose@v3.18.0
 	go install github.com/a-h/templ/cmd/templ@v0.2.543 
 	go install github.com/cosmtrek/air@v1.49.0 
 
-.PHONY: all build test clean
+install-tailwindcss:
+	if [ ! -f tailwindcss ]; then \
+		wget -q https://github.com/tailwindlabs/tailwindcss/releases/latest/download/tailwindcss-linux-x64 \
```
Comment (matheusgomes28) [inline Makefile:38]: We should probably be specific about the tailwind version here and do a checksum

File: views/not-found.templ [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package views
+
+import "github.com/matheusgomes28/urchin/common"
+
+templ makeNotFound() {
```
[the remark this answers is not in the document]
Comment (matheusgomes28) [reply views/not-found.templ:5]: can do later, not an issue

**Review by matheusgomes28 — APPROVED** [review]
LGTM

**Pull request merged**